quickwit-search/src/collector.rs

```diff
@@ -321,12 +321,13 @@ fn merge_leaf_responses(
         .sum();
     let failed_splits = leaf_responses
         .iter()
-        .flat_map(|leaf_response| leaf_response.failed_splits.iter().cloned())
+        .flat_map(|leaf_response| leaf_response.failed_splits.iter())
+        .cloned()
         .collect_vec();
-    let all_partial_hits: Vec<PartialHit> = leaf_responses
+    let all_partial_hits = leaf_responses
         .into_iter()
         .flat_map(|leaf_response| leaf_response.partial_hits)
-        .collect();
+        .collect_vec();
```

Comment:
```suggestion
        .collect();
```

Reply:
The `collect -> collect_vec()` change does not bring anything positive.

Writing the type explicitly from time to time in the code helps people reader from github for instance.

There is a reason why editor add an overlay with types.